internal/webhooks/machinedeployment.go

```diff
@@ -291,6 +292,10 @@ func (webhook *MachineDeployment) validate(oldMD, newMD *clusterv1.MachineDeploy
 		}
 	}
 
+	if newMD.Spec.MachineNamingStrategy != nil {
+		allErrs = append(allErrs, validateMDMachineNamingStrategy(newMD.Spec.MachineNamingStrategy, specPath.Child("machineNamingStrategy"))...)
```

Comment:
Could you kindly add add test coverage (same for MS)

Reply:
test coverage here in the webhook is still missing, right?

Reply:
tests are added